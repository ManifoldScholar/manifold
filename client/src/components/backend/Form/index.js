import HasMany from './HasMany';
import Radios from './Radios';
import TextArea from './TextArea';
import TextInput from './TextInput';
import MaskedTextInput from './MaskedTextInput';
import Upload from './Upload';
import Select from './Select';
import Save from './Save';
import Switch from './Switch';
import Date from './Date';
import HigherOrder from './HigherOrder';
import Hidden from './Hidden';
import AvatarBuilder from './AvatarBuilder/AvatarBuilder';
import setter from './setter';
import FieldGroup from './FieldGroup';

export default {
  HasMany,
  Hidden,
  Radios,
  TextArea,
  TextInput,
  MaskedTextInput,
  Upload,
  Select,
  Save,
  Switch,
  Date,
  AvatarBuilder,
  HigherOrder,
  FieldGroup,
  setter
};
