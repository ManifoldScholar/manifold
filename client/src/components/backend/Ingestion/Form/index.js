import Type from "./Type";
import Upload from "./Upload";
import Wrapper from "./Wrapper";

export default {
  Type,
  Upload,
  Wrapper
};
