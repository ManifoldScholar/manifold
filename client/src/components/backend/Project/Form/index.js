import AvatarBuilder from "./AvatarBuilder";
import ColorPicker from "./ColorPicker";
import Subjects from "./Subjects";

export default {
  AvatarBuilder,
  ColorPicker,
  Subjects
};
