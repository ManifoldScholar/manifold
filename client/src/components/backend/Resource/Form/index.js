import KindAttributes from './KindAttributes';
import Kind from './Kind';

export default {
  KindAttributes,
  Kind
};
