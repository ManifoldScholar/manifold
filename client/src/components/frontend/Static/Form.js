import React, { Component } from "react";

export default class StaticForm extends Component {
  static displayName = "Static.Form";

  render() {
    return (
      <div>
        {/*
          Background color can be adjusted by section and form
          appearance will follow suit. Options:
          blank (white)
          bg-neutral05 (light gray)
          bg-neutral90 (off black)
        */}
        <section>
          <div className="container">
            <h4 className="section-heading">Assorted form elements</h4>

            <div className="row-3-p">
              <div className="col-33">
                <div className="form-input form-error">
                  <label htmlFor="text-1">Text Entry Field</label>
                  <input
                    id="text-1"
                    type="text"
                    placeholder="Text Entry Field"
                  />
                  <i className="manicon manicon-stop" aria-hidden="true" />
                  <span className="form-error-message">something happened</span>
                </div>
              </div>

              <div className="col-33">
                <div className="form-input">
                  <label htmlFor="text-2">Text Entry Field</label>
                  <input
                    id="text-2"
                    type="text"
                    placeholder="Text Entry Field"
                  />
                </div>
              </div>

              <div className="col-33">
                <div className="form-input">
                  <label htmlFor="select-1">&nbsp;</label>
                  <div className="form-select">
                    <i
                      className="manicon manicon-caret-down"
                      aria-hidden="true"
                    />
                    <select id="select-1">
                      <option>Select Dropdown</option>
                      <option>Select an option</option>
                      <option>If you please</option>
                    </select>
                  </div>
                </div>
              </div>
            </div>

            <div className="row-3-p">
              <div className="col-66">
                <div className="form-input">
                  <label htmlFor="textarea-1">Editable text field</label>
                  <textarea
                    id="textarea-1"
                    placeholder="Start typing annotation here..."
                  />
                </div>
              </div>

              <div className="col-33">
                <div className="form-callout">
                  <header>
                    <i className="manicon manicon-bugle" aria-hidden="true" />
                    Notifications
                  </header>

                  <div className="form-input">
                    <label className="form-toggle radio annotated">
                      <input type="radio" name="notifications" />
                      <span className="toggle-indicator" />
                      <span className="toggle-label">Please Opt Me In</span>
                      <span className="toggle-note">
                        I have always been very upset when I am left out of the
                        group. This goes double for newsletters, promotional
                        offers, and targeted advertising.
                      </span>
                    </label>

                    <label className="form-toggle radio annotated">
                      <input type="radio" name="notifications" />
                      <span className="toggle-indicator" />
                      <span className="toggle-label">
                        {"None of your bee's wax"}
                      </span>
                      <span className="toggle-note">
                        I have several suggestions for where I would most like
                        to see your newsletter delivered, none of which include
                        my email address.
                      </span>
                    </label>
                  </div>
                </div>

                <div className="form-input">
                  <h4 className="form-input-heading">Checkboxes Vertical</h4>
                  {/* Radio buttons and checkboxes get wrapped in a toggle class */}
                  <label className="form-toggle checkbox">
                    <input type="checkbox" />
                    <span className="toggle-indicator" aria-hidden="true">
                      {/*
                        Checkboxes, and radio buttons require a .toggle-indicator in order
                        to display a checkbox or radio button. Additionally, checkboxes
                        require a "check" icon that is shown/hidden conditionally by
                        the state of the element.
                      */}
                      <i className="manicon manicon-check" />
                    </span>
                    <span className="toggle-label">Checkbox item</span>
                  </label>
                  <label className="form-toggle checkbox">
                    <input type="checkbox" />
                    <span className="toggle-indicator" aria-hidden="true">
                      <i className="manicon manicon-check" />
                    </span>
                    <span className="toggle-label">Checkbox item</span>
                  </label>
                  <label className="form-toggle checkbox">
                    <input type="checkbox" />
                    <span className="toggle-indicator" aria-hidden="true">
                      <i className="manicon manicon-check" />
                    </span>
                    <span className="toggle-label">Checkbox item</span>
                  </label>
                </div>

                <div className="form-input">
                  <h4 className="form-input-heading">Radios Vertical</h4>
                  <label className="form-toggle radio">
                    <input type="radio" name="test" />
                    <span className="toggle-indicator" aria-hidden="true">
                      {/* Radio Buttons don't have/require a check icon */}
                    </span>
                    <span className="toggle-label">Radio item</span>
                  </label>

                  <label className="form-toggle radio">
                    <input type="radio" name="test" />
                    <span className="toggle-indicator" aria-hidden="true" />
                    <span className="toggle-label">Radio item</span>
                  </label>

                  <label className="form-toggle radio">
                    <input type="radio" name="test" />
                    <span className="toggle-indicator" aria-hidden="true" />
                    <span className="toggle-label">Radio item</span>
                  </label>
                </div>
              </div>
            </div>

            <div className="row-3-p">
              <div className="col-66">
                <div className="form-input">
                  <h4 className="form-input-heading">Checkboxes Horizontal</h4>

                  <label className="form-toggle checkbox horizontal">
                    <input type="checkbox" />
                    <span className="toggle-indicator" aria-hidden="true">
                      <i className="manicon manicon-check" />
                    </span>
                    <span className="toggle-label">Checkbox item</span>
                  </label>

                  <label className="form-toggle checkbox horizontal">
                    <input type="checkbox" />
                    <span className="toggle-indicator" aria-hidden="true">
                      <i className="manicon manicon-check" />
                    </span>
                    <span className="toggle-label">Checkbox item</span>
                  </label>

                  <label className="form-toggle checkbox horizontal">
                    <input type="checkbox" />
                    <span className="toggle-indicator" aria-hidden="true">
                      <i className="manicon manicon-check" />
                    </span>
                    <span className="toggle-label">Checkbox item</span>
                  </label>
                </div>

                <div className="form-input">
                  <h4 className="form-input-heading">Checkboxes Horizontal</h4>

                  <label className="form-toggle radio horizontal">
                    <input type="radio" />
                    <span className="toggle-indicator" aria-hidden="true" />
                    <span className="toggle-label">Checkbox item</span>
                  </label>

                  <label className="form-toggle radio horizontal">
                    <input type="radio" />
                    <span className="toggle-indicator" aria-hidden="true" />
                    <span className="toggle-label">Checkbox item</span>
                  </label>

                  <label className="form-toggle radio horizontal">
                    <input type="radio" />
                    <span className="toggle-indicator" aria-hidden="true" />
                    <span className="toggle-label">Checkbox item</span>
                  </label>
                </div>
              </div>
            </div>

            <div className="row-3-p">
              <div className="col-33">
                {/* Use button-secondary-dull when button is disabled */}
                <div className="form-input">
                  <input
                    type="submit"
                    className="button-secondary-dull"
                    disabled
                  />
                  <input type="submit" className="button-secondary" />
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    );
  }
}
