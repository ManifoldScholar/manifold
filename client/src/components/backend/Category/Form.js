import React, { PureComponent, PropTypes } from 'react';
import { Form } from 'components/backend';
import { Form as FormContainer } from 'containers/backend';
import { textCategoriesAPI } from 'api';

export default class CategoryForm extends PureComponent {

  static displayName = "Category.Form";

  static propTypes = {
    projectId: PropTypes.string
  };

  render() {

    return (
      <FormContainer.Form
        {...this.props}
        name="backend-category-update"
        update={textCategoriesAPI.update}
        create={(model) => textCategoriesAPI.create(this.props.projectId, model) }
        className="form-secondary"
      >
        <Form.TextInput
          label="Title"
          focusOnMount
          name="attributes[title]"
          placeholder="What would you like to call this category?"
        />
        <Form.Save
          text={this.props.model ? "Update Category" : "Create Category" }
        />
      </FormContainer.Form>
    );
  }

}
