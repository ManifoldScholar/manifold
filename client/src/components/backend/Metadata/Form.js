import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { Form as FormContainer } from "containers/backend";
import { Form } from "components/backend";
import config from "../../../config";
import humps from "humps";

export default class FormMakers extends PureComponent {
  static displayName = "Metadata.Form";

  static propTypes = {
    model: PropTypes.object.isRequired
  };

  get attributes() {
    return this.props.model.attributes;
  }

  get metadataProperties() {
    if (!this.attributes || !this.attributes.metadataProperties) return [];
    return this.attributes.metadataProperties.sort();
  }

  get config() {
    return config.app.locale.metadata;
  }

  configFor(prop) {
    return this.config[prop];
  }

  labelize(prop) {
    return humps.decamelize(prop, { separator: " " });
  }

  configValueFor(prop, key) {
    const propConfig = this.configFor(prop);
    if (!propConfig || !propConfig.placeholder) return null;
    return propConfig[key];
  }

  placeholderFor(prop) {
    return this.configValueFor(prop, "placeholder");
  }

  instructionsFor(prop) {
    return this.configValueFor(prop, "instructions");
  }

  render() {
    return (
      <section>
        <FormContainer.Form {...this.props}>
          {this.metadataProperties.map((prop, i) => {
            const focus = i === 0;
            return (
              <Form.TextInput
                key={prop}
                focusOnMount={focus}
                placeholder={this.placeholderFor(prop)}
                instructions={this.instructionsFor(prop)}
                label={this.labelize(prop)}
                name={`attributes[metadata][${prop}]`}
              />
            );
          })}
          <Form.Save text="Save Metadata" />
        </FormContainer.Form>
      </section>
    );
  }
}

//
// <Form.TextInput
//   label="ISBN"
//   name="attributes[metadata][isbn]"
//   placeholder="Enter ISBN Number"
// />
// <Form.TextInput
// label="Publisher"
// name="attributes[metadata][publisher]"
// placeholder="Enter Publisher Name"
//   />
//   <Form.TextInput
// label="Place of Publication"
// name="attributes[metadata][placeOfPublication]"
// placeholder="Enter Place of Publication"
//   />
//   <Form.Date
// label="Publication Date"
// name="attributes[publicationDate]"
//   />
//   <Form.TextInput
// label="Digital Object Identifier (DOI)"
// name="attributes[metadata][doi]"
// placeholder="Enter DOI"
//   />
//   <Form.TextInput
// label="Series"
// name="attributes[metadata][series]"
// placeholder="Enter Series Name"
//   />
