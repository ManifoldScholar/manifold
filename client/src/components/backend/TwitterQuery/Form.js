import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { twitterQueriesAPI, requests } from "api";
import { Form as FormContainer } from "containers/backend";
import { Form } from "components/backend";

export default class TwitterQueryForm extends PureComponent {
  static displayName = "TwitterQuery.Form";

  static propTypes = {
    projectId: PropTypes.string,
    twitterQuery: PropTypes.object,
    successHandler: PropTypes.func,
    notificationScope: PropTypes.string,
    name: PropTypes.string.isRequired
  };

  static defaultProps = {
    notificationScope: "drawer"
  };

  handleSuccess = () => {
    if (!this.props.successHandler) return null;
    return this.props.successHandler();
  };

  render() {
    const pId = this.props.projectId;
    const twitterHelpUrl =
      "https://developer.twitter.com/en/docs/tweets/search/guides/standard-operators";
    const instructions = (
      <p className="instructions">
        Learn more about Twitter{" "}
        <a target="_blank" rel="noopener noreferrer" href={twitterHelpUrl}>
          search operators
        </a>.
      </p>
    );

    return (
      <FormContainer.Form
        model={this.props.twitterQuery}
        name={this.props.name}
        update={twitterQueriesAPI.update}
        create={model => twitterQueriesAPI.create(pId, model)}
        options={{ adds: requests.beTwitterQueries }}
        onSuccess={this.handleSuccess}
        className="form-secondary"
        notificationScope={this.props.notificationScope}
      >
        <Form.TextInput
          validation={["required"]}
          focusOnMount
          label="Query"
          name="attributes[query]"
          placeholder="Enter query"
          instructions={instructions}
        />
        <Form.Select
          name="attributes[resultType]"
          label="Fetch tweets by:"
          options={[
            { label: "Most Recent", value: "most_recent" },
            { label: "Most Popular", value: "popular" }
          ]}
        />
        <Form.Switch label="Active" name="attributes[active]" />
        <Form.Save text="Save Twitter Query" />
      </FormContainer.Form>
    );
  }
}
