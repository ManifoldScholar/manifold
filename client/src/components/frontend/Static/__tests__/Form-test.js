import React from 'react';
import renderer from 'react-test-renderer';
import Form from '../Form';

describe("Frontend.Static.Form component", () => {

  it('renders correctly', () => {
    const component = renderer.create(
      <Form  />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
