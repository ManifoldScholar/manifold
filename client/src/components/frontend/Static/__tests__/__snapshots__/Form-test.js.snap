// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frontend.Static.Form component renders correctly 1`] = `
<div>
  <section>
    <div
      className="container"
    >
      <h4
        className="section-heading"
      >
        Assorted form elements
      </h4>
      <div
        className="row-3-p"
      >
        <div
          className="col-33"
        >
          <div
            className="form-input form-error"
          >
            <label
              htmlFor="text-1"
            >
              Text Entry Field
            </label>
            <input
              id="text-1"
              placeholder="Text Entry Field"
              type="text"
            />
            <i
              aria-hidden="true"
              className="manicon manicon-stop"
            />
            <span
              className="form-error-message"
            >
              something happened
            </span>
          </div>
        </div>
        <div
          className="col-33"
        >
          <div
            className="form-input"
          >
            <label
              htmlFor="text-2"
            >
              Text Entry Field
            </label>
            <input
              id="text-2"
              placeholder="Text Entry Field"
              type="text"
            />
          </div>
        </div>
        <div
          className="col-33"
        >
          <div
            className="form-input"
          >
            <label
              htmlFor="select-1"
            >
              Â 
            </label>
            <div
              className="form-select"
            >
              <i
                aria-hidden="true"
                className="manicon manicon-caret-down"
              />
              <select
                id="select-1"
              >
                <option>
                  Select Dropdown
                </option>
                <option>
                  Select an option
                </option>
                <option>
                  If you please
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
      <div
        className="row-3-p"
      >
        <div
          className="col-66"
        >
          <div
            className="form-input"
          >
            <label
              htmlFor="textarea-1"
            >
              Editable text field
            </label>
            <textarea
              id="textarea-1"
              placeholder="Start typing annotation here..."
            />
          </div>
        </div>
        <div
          className="col-33"
        >
          <div
            className="form-callout"
          >
            <header>
              <i
                aria-hidden="true"
                className="manicon manicon-bugle"
              />
              Notifications
            </header>
            <div
              className="form-input"
            >
              <label
                className="form-toggle radio annotated"
              >
                <input
                  name="notifications"
                  type="radio"
                />
                <span
                  className="toggle-indicator"
                />
                <span
                  className="toggle-label"
                >
                  Please Opt Me In
                </span>
                <span
                  className="toggle-note"
                >
                  I have always been very upset when I am left out of the group. This goes double for newsletters, promotional offers, and targeted advertising.
                </span>
              </label>
              <label
                className="form-toggle radio annotated"
              >
                <input
                  name="notifications"
                  type="radio"
                />
                <span
                  className="toggle-indicator"
                />
                <span
                  className="toggle-label"
                >
                  None of your bee's wax
                </span>
                <span
                  className="toggle-note"
                >
                  I have several suggestions for where I would most like to see your newsletter delivered, none of which include my email address.
                </span>
              </label>
            </div>
          </div>
          <div
            className="form-input"
          >
            <h4
              className="form-input-heading"
            >
              Checkboxes Vertical
            </h4>
            <label
              className="form-toggle checkbox"
            >
              <input
                type="checkbox"
              />
              <span
                aria-hidden="true"
                className="toggle-indicator"
              >
                <i
                  className="manicon manicon-check"
                />
              </span>
              <span
                className="toggle-label"
              >
                Checkbox item
              </span>
            </label>
            <label
              className="form-toggle checkbox"
            >
              <input
                type="checkbox"
              />
              <span
                aria-hidden="true"
                className="toggle-indicator"
              >
                <i
                  className="manicon manicon-check"
                />
              </span>
              <span
                className="toggle-label"
              >
                Checkbox item
              </span>
            </label>
            <label
              className="form-toggle checkbox"
            >
              <input
                type="checkbox"
              />
              <span
                aria-hidden="true"
                className="toggle-indicator"
              >
                <i
                  className="manicon manicon-check"
                />
              </span>
              <span
                className="toggle-label"
              >
                Checkbox item
              </span>
            </label>
          </div>
          <div
            className="form-input"
          >
            <h4
              className="form-input-heading"
            >
              Radios Vertical
            </h4>
            <label
              className="form-toggle radio"
            >
              <input
                name="test"
                type="radio"
              />
              <span
                aria-hidden="true"
                className="toggle-indicator"
              />
              <span
                className="toggle-label"
              >
                Radio item
              </span>
            </label>
            <label
              className="form-toggle radio"
            >
              <input
                name="test"
                type="radio"
              />
              <span
                aria-hidden="true"
                className="toggle-indicator"
              />
              <span
                className="toggle-label"
              >
                Radio item
              </span>
            </label>
            <label
              className="form-toggle radio"
            >
              <input
                name="test"
                type="radio"
              />
              <span
                aria-hidden="true"
                className="toggle-indicator"
              />
              <span
                className="toggle-label"
              >
                Radio item
              </span>
            </label>
          </div>
        </div>
      </div>
      <div
        className="row-3-p"
      >
        <div
          className="col-66"
        >
          <div
            className="form-input"
          >
            <h4
              className="form-input-heading"
            >
              Checkboxes Horizontal
            </h4>
            <label
              className="form-toggle checkbox horizontal"
            >
              <input
                type="checkbox"
              />
              <span
                aria-hidden="true"
                className="toggle-indicator"
              >
                <i
                  className="manicon manicon-check"
                />
              </span>
              <span
                className="toggle-label"
              >
                Checkbox item
              </span>
            </label>
            <label
              className="form-toggle checkbox horizontal"
            >
              <input
                type="checkbox"
              />
              <span
                aria-hidden="true"
                className="toggle-indicator"
              >
                <i
                  className="manicon manicon-check"
                />
              </span>
              <span
                className="toggle-label"
              >
                Checkbox item
              </span>
            </label>
            <label
              className="form-toggle checkbox horizontal"
            >
              <input
                type="checkbox"
              />
              <span
                aria-hidden="true"
                className="toggle-indicator"
              >
                <i
                  className="manicon manicon-check"
                />
              </span>
              <span
                className="toggle-label"
              >
                Checkbox item
              </span>
            </label>
          </div>
          <div
            className="form-input"
          >
            <h4
              className="form-input-heading"
            >
              Checkboxes Horizontal
            </h4>
            <label
              className="form-toggle radio horizontal"
            >
              <input
                type="radio"
              />
              <span
                aria-hidden="true"
                className="toggle-indicator"
              />
              <span
                className="toggle-label"
              >
                Checkbox item
              </span>
            </label>
            <label
              className="form-toggle radio horizontal"
            >
              <input
                type="radio"
              />
              <span
                aria-hidden="true"
                className="toggle-indicator"
              />
              <span
                className="toggle-label"
              >
                Checkbox item
              </span>
            </label>
            <label
              className="form-toggle radio horizontal"
            >
              <input
                type="radio"
              />
              <span
                aria-hidden="true"
                className="toggle-indicator"
              />
              <span
                className="toggle-label"
              >
                Checkbox item
              </span>
            </label>
          </div>
        </div>
      </div>
      <div
        className="row-3-p"
      >
        <div
          className="col-33"
        >
          <div
            className="form-input"
          >
            <input
              className="button-secondary-dull"
              disabled={true}
              type="submit"
            />
            <input
              className="button-secondary"
              type="submit"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
